templates/block-block-js.mustache

```diff
@@ -1,33 +1,70 @@
 ( function( wp ) {
+	var registerBlockType = wp.blocks.registerBlockType;
 	var el = wp.element.createElement;
 	var __ = wp.i18n.__;
 
-	// Visit https://wordpress.org/gutenberg/handbook/block-api/ to learn about Block API
-	wp.blocks.registerBlockType( '{{namespace}}/{{slug}}', {
-		title: __( '{{title_ucfirst}}', '{{textdomain}}' ),
+	/**
+	 * Every block starts by registering a new block type definition.
+	 * The function takes two arguments, a block name and a block configuration object.
+	 * @see https://wordpress.org/gutenberg/handbook/block-api/
+	 */
+	registerBlockType( '{{namespace}}/{{slug}}', {
+		/**
+		 * This is the display title for your block, which can be translated with `i18n` functions.
+		 * The block inserter will show this name.
+		 */
+		title: __( '{{title_ucfirst}}' ),
 
 		{{#dashicon}}
+		/**
+		 * An icon property should be specified to make it easier to identify a block.
+		 * These can be any of WordPress’ Dashicons, or a custom svg element.
+		 */
 		icon: '{{dashicon}}',
 
 		{{/dashicon}}
+		/**
+		 * Blocks are grouped into categories to help users browse and discover them.
+		 * The core provided categories are `common`, `embed`, `formatting`, `layout` and `widgets`.
```

Comment:
This seems grammatically off to me. Depending on the intended meaning, I'd suggest either of these two:
- `The provided core categories`
- `The categories provided by core`